crates/proto/src/op/header.rs

```diff
@@ -82,6 +73,63 @@ pub enum MessageType {
     Response,
 }
 
+impl fmt::Display for MessageType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        let s = match self {
+            MessageType::Query => "QUERY",
+            MessageType::Response => "RESPONSE",
+        };
+
+        write!(f, "{}", s)
```

Comment:
I'd probably just `f.write_str()` in the match arms, seems simpler/more compact that way.

Reply:
great suggestion. for some reason I always forget to just use the interior methods on the formatter.